### DISCUSSION TITLE
SerialReadLine unable to read Sensirion SHT4X via arduino Uno

### DISCUSSION THREAD

**Discussion by chiyu1203** [bonsai-rx/bonsai#1982] [Q&A] [answered]
Dear Bonsai team, 
I was hoping to read the values from a thermo-sensor via arduino Uno. I used this [3rd-part library ](https://github.com/Sensirion/arduino-i2c-sht4x?tab=readme-ov-file) and was able to see the temperature and humidity value on Arduino's serial monitor. However, when I used SerialReadLine or CreateSerialPort, nothing shown up (I did change the baud rate to match the requirement written on the website). Did anyone have similar experience with this sensor or library before?
Below I have attached the workflow. Please feel free to let me know if you have any thought.
[thermo_sensor.zip](https://github.com/user-attachments/files/16746658/thermo_sensor.zip)

Best Regards
Chiyu

**Comment by chiyu1203** [comment, marked as answer]
I have managed to solve this issue. I think the key is to add delay in arduino code; and use both CreateSerialPort and SerialReadLine

**Reply by yqwang1** [reply]
Hi Do you mind sharing your working one for me as reference? I am also stuck there.

**Reply by chiyu1203** [reply]
Here you go! If you are testing a thermo-sensor, remember to raise the BaudRate in CreateSerialPort to the level that the sensor accepts.
[serial_readline_answer.bonsai.zip](https://github.com/user-attachments/files/18661953/serial_readline_answer.bonsai.zip)
